src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -821,7 +832,45 @@ private static Task ProcessUnknownFrameAsync()
             await OnEndStreamReceived();
         }
 
-        if (!_isMethodConnect && (_parsedPseudoHeaderFields & _mandatoryRequestPseudoHeaderFields) != _mandatoryRequestPseudoHeaderFields)
+        if (_isProtocolWebTransport)
+        {
+            // Requirement in the spec but this is handled by the Http/3 layer already, so leaving commented out for posterity
+            //if (Scheme != SchemeHttps)
+            //{
+            //    throw new Http3StreamErrorException("Scheme must be HTTPS." /*todo unhardcode*/, Http3ErrorCode.MessageError);
+            //}
+
+            if (!_isMethodConnect)
+            {
+                throw new Http3StreamErrorException("Method must be CONNECT." /*todo unhardcode*/, Http3ErrorCode.ProtocolError); // todo should this be a 400 response instead?
+            }
+
+            if (!ServerOptions.EnableWebTransport)
+            {
+                throw new Http3StreamErrorException("Server has WebTransport disabled. Please enable it prior to setting up WebTransport sessions." /*todo unhardcode*/, Http3ErrorCode.SettingsError);
+            }
+
+            if (!_parsedPseudoHeaderFields.HasFlag(PseudoHeaderFields.Authority) || !_parsedPseudoHeaderFields.HasFlag(PseudoHeaderFields.Path))
```

Comment:
This isn't conditional on the value of the :protocol header, only the presence of the header.
https://www.rfc-editor.org/rfc/rfc8441#section-4
>   On requests that contain the :protocol pseudo-header field, the
      :scheme and :path pseudo-header fields of the target URI (see
      [Section 5](https://www.rfc-editor.org/rfc/rfc8441#section-5)) MUST also be included.

Reply:
Can you please clarify what you mean? I am checking that the pseudoheaders have the flag set. Is this different from checking if the field is present?

Reply:
Nevermind, you addressed this on line 830